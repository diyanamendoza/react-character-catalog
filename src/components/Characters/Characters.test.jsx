import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Characters from './Characters';

it('should display a list of characters', () => {
  const { container } = render(
    <MemoryRouter>
      <Characters
        characters={[{
          image: 'https://placebear.com/200/300',
          name: 'Ted',
          species: 'bear',
          status: 'Alive',
        }]}
      />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
  screen.getByText(/Ted/i)
});
