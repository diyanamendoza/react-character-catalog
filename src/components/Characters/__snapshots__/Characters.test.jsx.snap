// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a list of characters 1`] = `
<div>
  <ul
    aria-label="characters"
    class="list"
  >
    <li
      class="item"
    >
      <a
        href="/characters/undefined"
      >
        <figure>
          <img
            alt="Ted"
            class="image"
            src="https://placebear.com/200/300"
          />
          <figcaption
            class="linkText"
          >
            Ted
          </figcaption>
        </figure>
      </a>
    </li>
  </ul>
</div>
`;
